src/plugins/json-schema-validator/index.js

```diff
@@ -0,0 +1,164 @@
+// JSON-Schema ( draf04 ) validator
+import JsonSchemaWebWorker from "./validator.worker.js"
+import PromiseWorker from "promise-worker"
+import debounce from "lodash/debounce"
+import swagger2Schema from "./swagger2-schema.js"
+import oas3Schema from "./oas3-schema"
+
+// Lazily created promise worker
+let _promiseWorker
+const promiseWorker = () => {
+  if(!_promiseWorker)
+    _promiseWorker = new PromiseWorker(new JsonSchemaWebWorker())
+  return _promiseWorker
+}
+
+export const addSchema = (schema, schemaPath=[]) => () => {
+  promiseWorker().postMessage({
+    type: "add-schema",
+    payload: {
+      schemaPath,
+      schema
+    }
+  })
+}
+
+// Figure out what schema we need to use ( we're making provision to be able to do sub-schema validation later on)
+// ...for now we just pick which base schema to use (eg: openapi-2-0, openapi-3.0, etc)
+export const getSchemaBasePath = () => ({ specSelectors }) => {
+  // Eg: [openapi-3.0] or [openapi-2-0]
+  // later on... ["openapi-2.0", "paths", "get"]
+  const isOAS3 = specSelectors.isOAS3 ? specSelectors.isOAS3() : false
+  const isSwagger2 = specSelectors.isSwagger2 ? specSelectors.isSwagger2() : false
+  const isAmbiguousVersion = isOAS3 && isSwagger2
+
+  // Refuse to handle ambiguity
+  if(isAmbiguousVersion)
+    return []
+
+  if(isSwagger2)
+    return ["openapi-2.0"]
+
+  if(isOAS3)
+    return ["openapi-3.0"]
+
+}
+
+
+export const setup = () => ({jsonSchemaValidatorActions}) => {
```

Comment:
shucks, I forgot about this... some code needs to call `.setup`
@shockey I was thinking we put a hook in src/index.js of swager-ui, an action that runs once after the system is built.

Reply:
@ponelat, we do give an `afterLoad` hook to plugins, would that work for you? docs: https://github.com/swagger-api/swagger-ui/blob/master/docs/customization/plugin-api.md#afterload

Reply:
Awesome, that'll do it! Its missing from this PR...
```js
return {
  // ...
  afterLoad: system => system.jsonSchemaValidator.setup(),
```